internal/common/kafka/kafka.go

```diff
@@ -18,26 +18,74 @@ var defaultTopic = "__consumer_offsets"
 // https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
 
 func NewProducer(config *viper.Viper) (*kafka.Producer, error) {
-	return kafka.NewProducer(&kafka.ConfigMap{
-		"bootstrap.servers":        config.GetString("kafka.bootstrap.servers"),
-		"request.required.acks":    config.GetInt("kafka.request.required.acks"),
-		"message.send.max.retries": config.GetInt("kafka.message.send.max.retries"),
-		"retry.backoff.ms":         config.GetInt("kafka.retry.backoff.ms"),
-	})
+
+	var producer *kafka.Producer
+	var err error
+	if config.Get("kafka.sasl.username") != nil {
+		producer, err = kafka.NewProducer(&kafka.ConfigMap{
```

Comment:
\<nitpick/\> Looks like there is some potential to reduce code duplication here. I.e. to first build ConfigMap with keys used in either case, then conditionally add auth properties.